src/_guides/language/effective-dart/design.md

```diff
@@ -908,37 +929,39 @@ install(id, destination) => ...
 {% endprettify %}
 
 Here, it's unclear what `id` is. A string? And what is `destination`? A string
-or a `File` object? Is this method synchronous or asynchronous?
+or a `File` object? Is this method synchronous or asynchronous? This is clearer:
 
 {:.good-style}
 <?code-excerpt "misc/lib/effective_dart/design_good.dart (type_annotate_public_apis)"?>
 {% prettify dart %}
 Future<bool> install(PackageId id, String destination) => ...
 {% endprettify %}
 
-With types, all of this is clarified.
-
-
-### DON'T specify a return type for a setter.
-
-{:.bad-style}
-<?code-excerpt "misc/lib/effective_dart/design_bad.dart (avoid_return_types_on_setters)"?>
-{% prettify dart %}
-void set foo(Foo value) { ... }
-{% endprettify %}
+In some cases, especially with constants, Dart can infer the type for you. In
+that case, it's fine to omit the annotation:
 
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (avoid_return_types_on_setters)"?>
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred)"?>
 {% prettify dart %}
-set foo(Foo value) { ... }
+const screenWidth = 640; // Infers "int" from 640.
 {% endprettify %}
 
+You may still wish to explicitly annotate even when inference would fill in the
+same type. If the inference relies on values or declarations from other
+libraries, you may want to annotate the type in *your* declaration so that a
+change to that other library doesn't silently change the type of your own API
+without you realizing.
 
-### PREFER type annotating private declarations.
+This guideline does *not* mean you can never use the dynamic type in a public
+signature. Sometimes `dynamic` is the best type. What it states is that when
+you want to use `dynamic`, you should do so *explicitly*.
```

Comment:
The "What it states ..." is a bit confusing. How about dropping that sentence and after "Sometimes `dynamic` is the best type" directing/linking the reader to the corresponding guideline?

Reply:
Simplified it to try to clarify. The intent of this paragraph isn't really about when you should use the dynamic type, it's just to try to make it clear that this guideline is about annotation, not dynamism.

After this PR lands, I may try to add some more guidance around when to you use the dynamic type (annotated or not).